resources/invoices/invoices.ts

```diff
@@ -0,0 +1,830 @@
+    options?: Core.RequestOptions,
+  ): Promise<Core.APIResponse<Invoice>>;
+  update(id: string, options?: Core.RequestOptions): Promise<Core.APIResponse<Invoice>>;
+  update(
+    id: string,
+    body: InvoiceUpdateParams | Core.RequestOptions = {},
+    options?: Core.RequestOptions,
+  ): Promise<Core.APIResponse<Invoice>> {
+    if (isRequestOptions(body)) {
+      return this.update(id, {}, body);
+    }
+    return this.patch(`/api/invoices/${id}`, { body, ...options });
+  }
+
+  /**
+   * list invoices
+   */
+  list(query?: InvoiceListParams, options?: Core.RequestOptions): Core.PagePromise<InvoicesPage>;
+  list(options?: Core.RequestOptions): Core.PagePromise<InvoicesPage>;
+  list(
+    query: InvoiceListParams | Core.RequestOptions = {},
+    options?: Core.RequestOptions,
+  ): Core.PagePromise<InvoicesPage> {
+    if (isRequestOptions(query)) {
+      return this.list({}, query);
+    }
+    return this.getAPIList('/api/invoices', InvoicesPage, { query, ...options });
+  }
+}
+
+export class InvoicesPage extends Page<Invoice> {}
+
+export interface Invoice {
+  /**
+   * The invoicer's contact details displayed at the top of the invoice.
+   */
+  contact_details: Array<Invoice.ContactDetails>;
+
+  /**
+   * The counterparty's billing address.
+   */
+  counterparty_billing_address: Invoice.CounterpartyBillingAddress | null;
+
+  /**
+   * The ID of the counterparty receiving the invoice.
+   */
+  counterparty_id: string;
+
+  /**
+   * The counterparty's shipping address where physical goods should be delivered.
+   */
+  counterparty_shipping_address: Invoice.CounterpartyShippingAddress | null;
+
+  created_at: string;
+
+  /**
+   * Currency that the invoice is denominated in. Defaults to `USD` if not provided.
+   */
+  currency: Shared.Currency | null;
+
+  /**
+   * A free-form description of the invoice.
+   */
+  description: string;
+
+  /**
+   * A future date by when the invoice needs to be paid.
+   */
+  due_date: string;
+
+  /**
+   * The URL of the hosted web UI where the invoice can be viewed.
+   */
+  hosted_url: string;
+
+  id: string;
+
+  /**
+   * The invoice issuer's business address.
+   */
+  invoicer_address: Invoice.InvoicerAddress | null;
+
+  /**
+   * This field will be true if this object exists in the live environment or false
+   * if it exists in the test environment.
+   */
+  live_mode: boolean;
+
+  /**
+   * A unique record number assigned to each invoice that is issued.
+   */
+  number: string;
+
+  object: string;
+
+  /**
+   * The ID of the internal account the invoice should be paid to.
+   */
+  originating_account_id: string;
+
+  /**
+   * The payment orders created for paying the invoice through the invoice payment
+   * UI.
+   */
+  payment_orders: Array<PaymentOrders.PaymentOrder>;
+
+  /**
+   * The URL where the invoice PDF can be downloaded.
+   */
+  pdf_url: string | null;
+
+  /**
+   * One of `draft`, `unpaid`, `payment_pending`, `paid`, `void`, or `closed`.
```

Comment:
@cleb11 I haven't approved/merged this PR yet since I was waiting for this to be synced. I thought it would be fixed automatically but I guess stainless hasn't picked up the OpenAPI change for this?

Reply:
ah rip, not sure. should we merge first so we can keep getting updates (since i doubt any ppl use invoice on sdk) and then follow up

Reply:
Yeah that seems reasonable